applications/serial_lte_modem/src/main.c

```diff
@@ -271,17 +267,32 @@ static void power_pin_callback_wakeup(const struct device *dev,
 	}
 	err = slm_at_host_power_on();
 	if (err) {
-		atomic_set(&callback_running, false);
-		LOG_ERR("Failed to power on uart: %d", err);
+		LOG_ERR("Failed to power on uart: %d. Resetting SLM.", err);
+		gpio_remove_callback(gpio_dev, &gpio_cb);
+		slm_reset();
```

Comment:
I've added a reset here as we discussed last week with @MarkusLassila. If we cannot turn the UART on, we are pretty much stuck anyway. It would be possible to handle this by enabling the powerpin but requires a bit more testing and not really sure whether it's worth it.